openfl-tutorials/interactive_api/MXNet_landmarks/workspace/MXNet_landmarks.ipynb

```diff
@@ -0,0 +1,573 @@
+   "id": "e6055103",
+   "metadata": {},
+   "source": [
+    "## Let's have a look at the results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ff804102",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from zipfile import ZipFile\n",
+    "\n",
+    "from kaggle.api.kaggle_api_extended import KaggleApi"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "37dc7f56",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not os.path.exists(\"./test\"):\n",
+    "    api = KaggleApi()\n",
+    "    api.authenticate()\n",
+    "    api.competition_download_file(\"facial-keypoints-detection\", \"test.zip\")\n",
+    "    with ZipFile(\"test.zip\", \"r\") as zipobj:\n",
+    "        zipobj.extractall(\"./test\")\n",
+    "    os.remove(\"test.zip\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "08fc3a7a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "last_model = fl_experiment.get_last_model()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "13f6cfd7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Test_Dir = \"./test/test.csv\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "796c8e37",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_data(path_to_csv_file):\n",
+    "    data_df = pd.read_csv(path_to_csv_file)\n",
+    "    data_df.fillna(method=\"ffill\", inplace=True)\n",
+    "    labels = data_df.drop(\"Image\", axis=1)\n",
+    "    imag, keypoints = [], []\n",
+    "    for i in range(data_df.shape[0]):\n",
+    "        img = data_df[\"Image\"][i].split(\" \")\n",
+    "        img = [\"0\" if x == \"\" else x for x in img]\n",
+    "        imag.append(img)\n",
+    "        y = labels.iloc[i, :]\n",
+    "        keypoints.append(y)\n",
+    "\n",
+    "    X = np.array(imag, dtype=\"float\").reshape(-1, 96, 96)\n",
+    "    y = np.array(keypoints, dtype=\"float\")\n",
+    "\n",
+    "    return X, y"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6ed1ce74",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_imgs, _ = get_data(Test_Dir)  # prepare test dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7cc6bed7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = plt.figure(figsize=(10, 10))\n",
+    "for i in range(9):\n",
+    "    ax = fig.add_subplot(3, 3, i + 1)\n",
+    "    in_for_net = (\n",
+    "        mx.nd.array([test_imgs[i + 1]]).expand_dims(axis=1).as_in_context(mx.cpu())\n",
+    "    )\n",
+    "    pred = last_model(in_for_net)[0].asnumpy().reshape(-1, 2)\n",
+    "    ax.imshow(test_imgs[i + 1], cmap=\"gray\")\n",
+    "    x_cords = pred[:, 0]\n",
+    "    y_cords = pred[:, 1]\n",
+    "    plt.scatter(x_cords, y_cords)\n",
+    "plt.show()"
```

Comment:
Suggestion:
Just to add legend
```suggestion
    "    plt.scatter(x_cords, y_cords, label='Predicted keypoint')\n",
    "plt.legend(loc=1, prop={'size': 12})\n",
    "plt.show()"
```
Should looks like:
![image](https://user-images.githubusercontent.com/89856652/156024818-3dc04784-a043-41af-b808-37eb2fbd0465.png)